packages/fuels-core/src/parse.rs

```diff
@@ -0,0 +1,191 @@
+use fuels_types::{
+    errors::Error,
+    param_types::{EnumVariants, ParamType},
+    Property,
+};
+use std::str::FromStr;
+
+/// Turns a JSON property into ParamType
+pub fn parse_param_type_from_property(prop: &Property) -> Result<ParamType, Error> {
```

Comment:
Perhaps we can transform this into a  `TryFrom<Property>` implementation for `ParamType`. 

Seems like the idiomatic approach to type conversions that might fail? Then we could write:
```rust
let param_type: ParamType = prop.try_into()?;
```

Reply:
Yeah I was thinking about that too, it would make more sense, but I was kind of leaving it for "later" refactors. But now that you mention it nothing prevents from doing it right now.

Reply:
So I'm going to do all this + more in another PR, this one is already getting too big!